src/components/sections.tsx

```diff
@@ -9,11 +10,15 @@ export interface TOCEntry {
     children?: TOCEntry[];
 }
 
-export const ContentsContext = createContext<[number, (entry: TOCEntry, depth: number) => void, boolean, React.MutableRefObject<TOCEntry[]> | null]>([
+export const ContentsContext = createContext<
+    [number, (entry: TOCEntry, depth: number) => void, boolean, React.MutableRefObject<TOCEntry[]> | null]
+>([
     0,
-    (entry: TOCEntry, depth: number) => {},
+    (entry: TOCEntry, depth: number) => {
+        // ? Fill this out?
```

Comment:
This method of automatically generating a table of contents based on the page contents probably has a better solution. In this case, it is expected to use the `useTableOfContents` hook to retrieve a function as the provider value, and then wrap the sections that will be generating the table with the context provider itself. There is no default behavior for the context, it only affects the provided ToC ref that's also provided by the hook.

Reply:
Sounds good! I just really put that there because the linter was complaining about it being an empty arrow function. Also I'm working on resolving the conflicts right now